v8py/utils.cpp

```diff
@@ -0,0 +1,47 @@
+
+#include "utils.h"
+#include "v8py.h"
+#include "jsobject.h"
+#include "convert.h"
+
+using namespace v8;
+
+PyObject *construct_new_object(PyObject *self, PyObject *args) {
+
+    int argc = (int)PyTuple_GET_SIZE(args);
```

Comment:
Why not use PyArg_ParseTuple?

Reply:
I indeed tried to use `PyArg_ParseTuple` with `O!` but it gave me hard time (the js_function* reference it gives was corrupt leading to SIGSEGV) so I ended up with simple `PyTuple_GET_ITEM` / `PyObject_TypeCheck`

Reply:
strange! oh well.

Reply:
Literally the code I've tried:
```
    js_function *function;
    if (PyArg_ParseTuple(args, "O!", &js_function_type, &function) < 0) {
        PyErr_SetString(PyExc_TypeError, "First argument must be a constructor function.");
        return NULL;
    }
```
`function` reference after this execution was way dead.